src/features/annotation/components/task-detail/page-client.tsx

```diff
@@ -75,6 +79,83 @@ export function AnnotationTaskDetailPageClient({
     }
   }, [page, isFetching, isLoading]);
 
+  const handleAnnotationSuccess = useCallback(async () => {
+    try {
+      const cachedAllAnnotations = queryClient.getQueriesData({
+        queryKey: annotationKeys.taskAnnotations(taskId),
+        exact: false,
+      });
+
+      let firstPendingId: number | null = null;
+      let cachedIndex = -1;
+
+      for (const [, data] of cachedAllAnnotations) {
+        if (data && typeof data === 'object' && 'items' in data) {
+          const items = (data as any).items;
+          cachedIndex = items.findIndex((item: any) => item.status === 'PENDING');
+          if (cachedIndex !== -1) {
+            firstPendingId = items[cachedIndex].id;
+            break;
+          }
+        }
+      }
+
+      if (firstPendingId !== null && cachedIndex !== -1) {
+        setPage(cachedIndex + 1);
+        return;
+      }
+
+      const pendingAnnotations = await getAnnotations({
+        taskId,
+        page: 1,
+        limit: 1,
+        status: 'PENDING',
+      });
+
+      if (pendingAnnotations.items && pendingAnnotations.items.length > 0) {
+        firstPendingId = pendingAnnotations.items[0].id;
+        
+        let found = false;
+        let currentBatch = 1;
+        const batchSize = 50;
+        
+        while (!found && currentBatch <= 10) {
+          const batch = await getAnnotations({
+            taskId,
+            page: currentBatch,
+            limit: batchSize,
+          });
+          
+          const indexInBatch = batch.items.findIndex(
+            (annotation) => annotation.id === firstPendingId
+          );
+          
+          if (indexInBatch !== -1) {
+            const overallIndex = (currentBatch - 1) * batchSize + indexInBatch;
+            setPage(overallIndex + 1);
+            found = true;
+          } else if (!batch.hasMore) {
+            if (hasMore) setPage(prev => prev + 1);
+            break;
+          }
+          
+          currentBatch++;
+        }
+        
+        if (!found && hasMore) {
```

Comment:
Why do you need to run through each page? I wonder if there is a simpler way for you to query which is the next pending ID, and then just navigate to that one. Think about whether you can use the response from the backend to derive which page you should navigate to as opposed to running through a while loop.